test-integration/certificates/generate_leaf_certs.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Set variables
+ROOT_CA_CERT="generated_assets/bacalhau_test_root_ca.crt"
+ROOT_CA_KEY="generated_assets/bacalhau_test_root_ca.key"
```

Comment:
Are these used anywhere? Is the intention to pass them to bacalhau at some point?

Reply:
The CA is being injected into every image truststore. Potentially all the certificates used by this bacalhau deployment will be signed by it. Currently we only use it to sign the Docker Registry TLS cert.

The key on our hand, is needed because we need to sign any new certificates with it.